fe/fe-core/src/test/java/com/starrocks/transaction/ConcurrentTxnTest.java

```diff
@@ -0,0 +1,183 @@
+            stmt.close();
+            connection.close();
+        }
+    }
+
+    class TableLoad {
+        String db;
+        String table;
+        int id;
+        int numTablet;
+        int replicationNum;
+        int loadIntervalMs;
+
+        TableLoad(String db, int id, int numTablet, int replicationNum, int loadIntervalMs) {
+            this.db = db;
+            this.table = "table_" + id;
+            this.id = id;
+            this.numTablet = numTablet;
+            this.replicationNum = replicationNum;
+            this.loadIntervalMs = loadIntervalMs;
+        }
+
+        void createTable() throws SQLException {
+            runSql(db,
+                    "create table " + table +
+                            " ( pk bigint NOT NULL, v0 string not null) primary KEY (pk) DISTRIBUTED BY HASH(pk) BUCKETS " +
+                            numTablet + " PROPERTIES(\"replication_num\" = \"" + replicationNum +
+                            "\", \"storage_medium\" = \"SSD\");");
+        }
+
+        boolean hasTask(int tsMs) {
+            return tsMs % loadIntervalMs == 0;
+        }
+
+        void loadOnce() throws SQLException {
+            runSql(db, "insert into " + table + " values (1,\"1\"), (2,\"2\"), (3,\"3\");");
+        }
+    }
+
+    class DBLoad {
+        String db;
+        int numTable;
+        List<TableLoad> tableLoads;
+        volatile Exception error;
+
+        int finishedTask = 0;
+
+        DBLoad(String db, int numTable) {
+            this.db = db;
+            this.numTable = numTable;
+        }
+
+        void run(int numThread, int runSeconds) {
+            try {
+                runSql(null, "create database " + db);
+            } catch (SQLException e) {
+                Assert.fail(e);
+            }
+            ThreadPoolExecutor executor
+                    = (ThreadPoolExecutor) Executors.newFixedThreadPool(numThread);
+            tableLoads = new ArrayList<>();
+            Random r = new Random(0);
+            int scheduleIntervalMs = 200;
+            for (int i = 0; i < numTable; i++) {
+                int numTablet = r.nextInt(16) + 1;
+                int replicationNum = r.nextInt(3) + 1;
+                int loadIntervalMs = (r.nextInt(40) + 1) * scheduleIntervalMs;
+                TableLoad tableLoad = new TableLoad(db, i, numTablet, replicationNum, loadIntervalMs);
+                tableLoads.add(tableLoad);
+                executor.submit(() -> {
+                    try {
+                        tableLoad.createTable();
+                    } catch (SQLException e) {
+                        error = e;
+                    }
+                });
+                if (error != null) {
+                    Assert.fail(error);
+                }
+            }
+            List<Future<?>> futures = new ArrayList<Future<?>>();
+            finishedTask = 0;
+            long startTs = System.nanoTime();
+            int runInterval = runSeconds * 1000 / scheduleIntervalMs;
```

Comment:
The runInterval equals to 50, the number of tableLoads is 1000. The mulitple result is 50000 publish tasks. It not correponds to the commit message.